packages/optimizely-sdk/lib/optimizely/index.ts

```diff
@@ -110,6 +114,12 @@ export default class Optimizely {
     this.isOptimizelyConfigValid = config.isValidInstance;
     this.logger = config.logger;
 
+    let defaultDecideOptions = config.defaultDecideOptions ?? [];
+    if (!Array.isArray(defaultDecideOptions)) {
+      this.logger.log(LOG_LEVEL.DEBUG, sprintf(LOG_MESSAGES.INVALID_DECIDE_OPTIONS, MODULE_NAME));
```

Comment:
I believe the logger has format string support built in, there's no need to call `sprintf` here. Try this:
```suggestion
      this.logger.debug(LOG_MESSAGES.INVALID_DECIDE_OPTIONS, MODULE_NAME);
```
A lot of the log statements in this model do not use this because they predate this support being added in the logger.

Reply:
I think this would require a bit more work than just changing it as shown above. Right now we have logger as `ConsoleLogHandler`, where it would require changing it to `OptimizelyLogger` to support built-in format string. I will create a separate ticket to have it updated.